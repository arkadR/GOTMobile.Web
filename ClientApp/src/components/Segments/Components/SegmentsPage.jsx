import React, { useState, useEffect } from "react";
import SegmentsTable from "./SegmentsTable";
import SegmentForm from "./SegmentForm";
import SegmentsStore from "../Stores/SegmentsStore";
import { loadSegments } from "../Actions/SegmentActions";

export default function SegmentsPage(props) {
  const [segments, setSegments] = useState(SegmentsStore.getSegments());
  const [points, setPoints] = useState(SegmentsStore.getPoints());
  const [selectedSegment, setSelectedSegment] = useState(
    SegmentsStore.getSelectedSegment()
  );

  useEffect(() => {
    SegmentsStore.addChangeListener(() => {
      setSegments(SegmentsStore.getSegments());
      setPoints(SegmentsStore.getPoints());
      setSelectedSegment(SegmentsStore.getSelectedSegment());
    });
    if (SegmentsStore.getSegments().length === 0) {
      loadSegments();
    }
  });

  return (
    <>
      <SegmentsTable
        segments={segments}
        points={points}
        selectedSegmentId={selectedSegment ? selectedSegment.id : -1}
      />
      {selectedSegment && (
        <SegmentForm points={points} selectedSegment={selectedSegment} />
      )}
    </>
  );
}
