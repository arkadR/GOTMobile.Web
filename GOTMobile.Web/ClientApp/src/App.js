import React, { Component } from "react";
import { Route } from "react-router";
import Layout from "./components/Layout";
import { HomePage } from "./components/HomePage";
import SegmentsPage from "./Segments/Components/SegmentsPage";

import "./custom.css";

const navItems = [
  {
    link: "/segments",
    title: "Odcinki"
    // icon: <p></p>
  },
  {
    link: "/points",
    title: "Punkty"
    // icon: <p></p>
  }
];

export default class App extends Component {
  static displayName = App.name;

  render() {
    return (
      <>
        <Layout navItems={navItems}>
          <Route exact path="/" component={HomePage} />
          <Route path="/segments" component={SegmentsPage} />
        </Layout>
      </>
    );
  }
}
