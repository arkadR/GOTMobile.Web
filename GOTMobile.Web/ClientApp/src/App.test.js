import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";
import { MemoryRouter } from "react-router-dom";
import App from "./App";
import { create, act } from "react-test-renderer";

import DeleteSegmentDialog from "./Segments/Components/DeleteSegmentDialog";
import SegmentsPage from "./Segments/Components/SegmentsPage";
import SegmentsForm from "./Segments/Components/SegmentForm";
import { Button, TableRow } from "@material-ui/core";

const noSegmentSelected = -1;

it("renders without crashing", async () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
    div
  );
  await new Promise(resolve => setTimeout(resolve, 1000));
});

describe("DeleteSegmentDialog component", () => {
  test("Matches the snapshot", () => {
    const dialog = create(<DeleteSegmentDialog />);
    expect(dialog.toJSON()).toMatchSnapshot();
  });
});

describe("Add segment button clicked", () => {
  test.only("Selecting same segment twice", async () => {
    const page = create(<SegmentsPage />);
    const instance = page.root;
    const button = instance.findByType(Button);
    act(() => button.props.onClick());
    // const form = instance.findByType(SegmentsForm);

    // expect(form.props.selectedSegment).not.toBe(-1);
  });

  test("Selecting same segment twice", async () => {
    //Arrange
    const page = create(<SegmentsPage />);
    const instance = page.root;
    const tableRow = instance.find(
      node => (node.id = "enhanced-table-checkbox-2")
    );
    const form = instance.findByType(SegmentsForm);

    //Act, Assert
    act(() => ReactTestUtils.Simulate.click(tableRow));
    expect(form.props.selectedSegment).not.toBe(noSegmentSelected);

    act(() => ReactTestUtils.Simulate.click(tableRow));
    await waitForDomChange();
    expect(form.props.selectedSegment).toBe(noSegmentSelected);
  });
});
