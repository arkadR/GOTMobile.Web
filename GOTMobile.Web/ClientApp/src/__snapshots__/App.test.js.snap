// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteSegmentDialog component Matches the snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Open alert dialog
    </span>
  </button>
</div>
`;
